tests/contrib/kafka/test_kafka.py

```diff
@@ -101,6 +104,22 @@ def consumer(tracer, kafka_topic):
     _consumer.close()
 
 
+@pytest.fixture
+def non_auto_commit_consumer(tracer, kafka_topic):
```

Comment:
The short answer is "yes", the longer answer is this split should help split up tests so we aren't having to use subprocesses to try different Agent configurations anymore (i.e. for datastreams we may be able to get away with a single suite of tests with `DD_DATA_STREAMS_ENABLED` set instead of spawning a subprocess per test as we do now).

With that said, the immediate answer is "I'm refactoring in parts, so the tests will come later"